src/main/java/com/uber/cadence/internal/sync/CancellationScopeImpl.java

```diff
@@ -73,6 +74,13 @@ private static void popCurrent(CancellationScopeImpl expected) {
     setParent(parent);
   }
 
+  public CancellationScopeImpl(
+      boolean ignoreParentCancellation, Functions.Proc1<CancellationScope> proc) {
+    this.detached = ignoreParentCancellation;
+    this.runnable = () -> proc.apply(this);
+    setParent(current());
```

Comment:
Do need another constructor to pass parent explicitly like other constructor?

Reply:
The only place where the parent is used is when a new child thread is created and the thread cancellation scope should be connected to the parents thread cancellation scope. So no need for the parent parameter when Proc1 one is passed.